lib/logstash/codecs/cef.rb

```diff
@@ -279,14 +281,15 @@ def handle(data, &block)
     message = unprocessed_data
 
     # Try and parse out the syslog header if there is one
-    if event.get('cefVersion').include? ' '
-      split_cef_version= event.get('cefVersion').rpartition(' ')
+    if (cef_version = event.get('cefVersion')).include?(' ')
+      split_cef_version = cef_version.rpartition(' ')
       event.set('syslog', split_cef_version[0])
-      event.set('cefVersion',split_cef_version[2])
+      event.set('cefVersion', split_cef_version[2])
     end
 
     # Get rid of the CEF bit in the version
-    event.set('cefVersion', event.get('cefVersion').sub(/^CEF:/, ''))
+    cef_version = event.get('cefVersion')
+    event.set('cefVersion', cef_version[CEF_PREFIX.length..-1]) if cef_version.start_with?(CEF_PREFIX)
```

Comment:
not sure if the performance benefit here outweights the readability impact. maybe we can prepare the code for the future with
```suggestion
    # make this a constant 
    if Gem::Requirement.new(">= 2.5.0").satisfied_by? Gem::Version.new(RUBY_VERSION)
      event.set('cefVersion', cef_version.delete_prefix(CEF_PREFIX))
    else
      event.set('cefVersion', cef_version[CEF_PREFIX.length..-1]) if cef_version.start_with?(CEF_PREFIX)
    end
```

Reply:
if you feel like readability has suffered than it might not be worth the hustle ...
after all its an anchored regexp - there's mostly the overhead of setting "globals" `$~` etc.

will give it one more try of extracting your suggestion into a method and see how that looks ...

Reply:
updated, please take a stab. if it reads ugly :game_die: